### PR TITLE
Generate errors when referencing an Executable project with different SelfContained value

### LINKED ISSUE #15117: Generate error if SelfContained Exe references framework dependent Exe, or vice versa
With #14488, we will support Exe projects referencing Exe projects, and both apps should be runnable from the referencing project's output folder.  However, the apps won't run correctly work if one of them is self-contained and the other isn't (due to how hostfxr works).  So we should generate an error if a self-contained Exe references a framework dependent Exe, or vice versa.

This error should be suppressible in case there are Exe projects that want to reference other Exe projects where SelfContained doesn't match, but they don't want a runnable output of the referenced project, they just want to call APIs from that assembly.  Also adding this error message will be a potentially breaking change that we should document.

### PR DESCRIPTION
**Pull request by dsplaisted**
Fixes #15117

Requires changes to MSBuild in dotnet/msbuild#5994 before this will work.

### CODE REVIEW

**Comment by wli3** [comment]
@dsplaisted that's the other PR need review? Although tests failed

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets [hunk 1]
```diff
@@ -1029,6 +1029,31 @@ Copyright (c) .NET Foundation. All rights reserved.
...
+
+  <ItemGroup>
+    <AdditionalTargetFrameworkInfoProperty Include="SelfContained"/>
+    <AdditionalTargetFrameworkInfoProperty Include="_IsExecutable"/>
+  </ItemGroup>
+
+  <UsingTask TaskName="ValidateExecutableReferences" AssemblyFile="$(MicrosoftNETBuildTasksAssembly)" />
+
+  <Target Name="ValidateExecutableReferences"
+          AfterTargets="_GetProjectReferenceTargetFrameworkProperties"
```
Comment (wli3) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets:1046]: Cannot make this a dependOn?

**Review by wli3 — APPROVED** [review]

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets (hunk 1 above)
Comment (dsplaisted) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets:1046]: No, `DependsOnTargets` and `AfterTargets` mean different things: https://github.com/MicrosoftDocs/visualstudio-docs/issues/2994#issuecomment-480017831

**Comment by dsplaisted** [comment]
@marcpopMSFT @KathleenDollard FYI for breaking change documentation

**Pull request merged**